Restrict cookie length

Hugh-OBrien opened the pull request:
Adds the ability to restrict the max length of a param to store as a cookie. This prevents arbitrarily long strings being stored as cookies and potentially being stored by an application at some point. 

I've set the default value of `max_length` to 30 and updated the minor version to reflect this change.

Commit:
added max_length option with default and tests

Commit:
update minor version

jdleesmiller commented on lib/rack/param_to_cookie.rb:
```diff
@@ -33,6 +34,9 @@ def call env
         params_value = req.params[param] rescue nil
 
+        # validate the length of the value
+        next if req.params[param].length > options[:max_length] rescue nil
```
Is there a reason not to use `params_value` here? It looks like we've already handled potential errors above, so we should now be fairly confident that we have either a string or `nil`.

Also, as written this will skip the rest of the function if the request parameter is invalid. I don't see a reason that we couldn't continue with `params_value` `nil` if it was invalid, and still set the value in the env from the cookie?

So... could this line just be
```
params_value = nil if params_value && params_value.length > options[:max_length]
```
?

Hugh-OBrien replied:
>Is there a reason not to use params_value here? It looks like we've already handled potential errors above, so we should now be fairly confident that we have either a string or nil.

Nope

>Also, as written this will skip the rest of the function if the request parameter is invalid. I don't see a reason that we couldn't continue with params_value nil if it was invalid, and still set the value in the env from the cookie?

Good point, lets go with your line

jdleesmiller commented on lib/rack/param_to_cookie/version.rb:
```diff
@@ -1,7 +1,7 @@
   class ParamToCookie
     VERSION_MAJOR = 2
-    VERSION_MINOR = 0
+    VERSION_MINOR = 1
```
I think it could be argued that this is a breaking change under [semver](http://semver.org/): if you were using it to set really big cookies, then upgrading to this version would cause it to quietly drop them. If it is, it should be a version major.

Hugh-OBrien replied:
Yes, I think this would be true no matter what default value we set for `max_value`. I think it's better to have a sensible default so yes this will be a breaking change.

jdleesmiller commented on lib/rack/param_to_cookie.rb:
```diff
@@ -19,6 +19,7 @@ def initialize app, param_cookies
         options[:env_name] ||= param
         options[:ttl] ||= 60*60*24*30 # 30 days
         options[:set_cookie_options] ||= {}
+        options[:max_length] ||= 30 # characters
```
I think we should definitely set a default max length, but 30 strikes me as a bit short --- UUIDs, SHA1 hashes, etc., are all longer than this. What would you say to 64 characters?

Hugh-OBrien replied:
64 sounds good to me! 30 was a pretty arbitrary choice on my end

jdleesmiller reviewed (changes requested):
OK, looks like a good start.

One other request: could you also update the README to mention this new option? Maybe a sentence at the end of the Usage section?

Commit:
Review fixes
- Store nil when max value is exceeded
- Max length to 64 chars
- Readme for the max_length
- Update major version number

Hugh-OBrien commented:
All good points. I've addressed them and added some extra detail to the detailed readme example.
